.github/scripts/sink_cache.py

```diff
@@ -0,0 +1,235 @@
+
+basicConfig(level=logging.INFO)
+logger = getLogger()
+
+
+def gh_api(url: str, method: Optional[str] = "GET"):
+    extra_args = [f"--header='Accept: application/json'"]
+    if method is not None:
+        extra_args.append(f"--method={method}")
+    cmd_args = ["gh", "api", *extra_args, url]
+    logger.debug(f"Will execute cmd `{' '.join(cmd_args)}`")
+    ret = subprocess.check_output(cmd_args)
+    return json.loads(ret)
+
+
+class CacheEntry:
+    id: int
+    key: str
+    ref: str
+    version: str
+    last_accessed_at: datetime
+    created_at: datetime
+    size_in_bytes: int
+
+    def __init__(
+        self,
+        id: int,
+        key: str,
+        ref: str,
+        version: str,
+        last_accessed_at: datetime,
+        created_at: datetime,
+        size_in_bytes: int,
+    ) -> None:
+        self.id = id
+        self.key = key
+        self.ref = ref
+        self.version = version
+        self.last_accessed_at = last_accessed_at
+        self.created_at = created_at
+        self.size_in_bytes = size_in_bytes
+
+    def from_dict(dict: Dict[str, Any]) -> "CacheEntries":
+        # Datetime format for `2019-01-24T22:45:36.000Z``
+
+        def get_iso_datetime(raw: str) -> datetime:
+            # `fromisoformat` don't support:
+            # - nano second precision, so we need to remove it, so we need to remove it.
+            # - `Z` timezone specifier for the UTC timezone, so we need to replace it.
+            #
+            # YYYY-MM-DDTHH:MM:SS.ffffff
+            log = logger.getChild(get_iso_datetime.__name__)
+            index_z = raw.index("Z", len(raw) - 1)
+            # index_timezone = raw.index("+", len(raw) - 5)
+            if index_z != -1:
+                # Remove nanosecond
+                trim_index = min(26, index_z)
+                trimmed_raw = raw[:trim_index]
+                log.debug(f"raw={raw}, trimmed_raw={trimmed_raw}, trim_index={trim_index}")
+                # Replace the `Z` UTC identifier by `+00:00`
+                dt = datetime.fromisoformat(trimmed_raw + "+00:00")
+            else:
+                raise NotImplementedError()
+            return dt
+
+        last_accessed_at = get_iso_datetime(dict["last_accessed_at"])
+        create_at = get_iso_datetime(dict["created_at"])
+
+        return CacheEntry(
+            id=int(dict["id"]),
+            key=dict["key"],
+            ref=dict["ref"],
+            version=dict["version"],
+            last_accessed_at=last_accessed_at,
+            created_at=create_at,
+            size_in_bytes=int(dict["size_in_bytes"]),
+        )
+
+
+class CacheEntries:
+    total_count: int
+    actions_caches: Dict[int, CacheEntry]
+
+    def __init__(self, total_count: int, actions_caches: Dict[int, CacheEntry]) -> None:
+        self.total_count = total_count
+        self.actions_caches = actions_caches
+
+    def __repr__(self) -> str:
+        return f"{CacheEntries.__name__}({self.total_count}, {repr(self.actions_caches)})"
```

Comment:
you can do `f"{x:r}` instead of `f"{repr(x)}`

instead of using `CacheEntries.__name__`, we do `type(self).__name__` in repr. This is because otherwise a child class would have the wrong name here.

and finally you would benefit from using `@dataclasses.dataclass` given it would provide you with the repr for free ;-)

Reply:
the classes now use `@dataclass`